Resolved the remaining uses of WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN in /WebKit/ in Apple ports

geoffreygaren opened the pull request:
#### cbab3e78a6a1b42a42b5d1a1560b97b11286e7a3
<pre>
Resolved the remaining uses of WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN in /WebKit/ in Apple ports
<a href="https://bugs.webkit.org/show_bug.cgi?id=282730">https://bugs.webkit.org/show_bug.cgi?id=282730</a>
<a href="https://rdar.apple.com/139395062">rdar://139395062</a>

Reviewed by Chris Dumez.

You can now audit WebKit by searching for the regex

    WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN(?! //.*port)

and it should turn up no results.

Fixed one or two remaining unsafe buffer uses.

Added a WTF_IGNORE_WARNINGS_IN_THIRD_PARTY_CODE_BEGIN/_END to distinguish code
that we&apos;re skipping because it&apos;s not a part of WebKit. (Clang treats system
headers similarly.)

Added comments to annotate port-specific uses of
WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN to facilitate regex audit.

Removed some markers that skipped constructs that are now demonstrably safe.

* Source/WTF/wtf/Compiler.h:
* Source/WebKit/GPUProcess/EntryPoint/Cocoa/XPCService/GPUServiceEntryPoint.mm:
(GPU_SERVICE_INITIALIZER):
* Source/WebKit/GPUProcess/webrtc/LibWebRTCCodecsProxy.mm:
* Source/WebKit/NetworkProcess/cache/NetworkCacheDataGLib.cpp:
(WebKit::NetworkCache::concatenate):
* Source/WebKit/NetworkProcess/soup/WebSocketTaskSoup.cpp:
(WebKit::WebSocketTask::WebSocketTask):
* Source/WebKit/NetworkProcess/webrtc/LibWebRTCSocketClient.h:
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCProvider.cpp:
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCProvider.h:
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCTCPSocketCocoa.mm:
* Source/WebKit/Platform/IPC/JSIPCBinding.h:
* Source/WebKit/Platform/IPC/StreamConnectionEncoder.h:
* Source/WebKit/Platform/IPC/glib/ArgumentCodersGlib.h:
* Source/WebKit/Platform/IPC/unix/ConnectionUnix.cpp:
* Source/WebKit/Platform/IPC/unix/UnixMessage.h:
(IPC::UnixMessage::UnixMessage):
* Source/WebKit/Shared/API/c/skia/WKImageSkia.h:
* Source/WebKit/Shared/API/glib/WebKitURIRequest.cpp:
(webkit_uri_request_set_uri):
* Source/WebKit/Shared/EntryPointUtilities/Cocoa/XPCService/XPCServiceEntryPoint.h:
* Source/WebKit/Shared/EntryPointUtilities/Cocoa/XPCService/XPCServiceEntryPoint.mm:
(WebKit::setJSCOptions):
* Source/WebKit/Shared/RTCNetwork.h:
* Source/WebKit/Shared/RTCPacketOptions.h:
* Source/WebKit/Shared/RTCWebKitEncodedFrameInfo.h:
* Source/WebKit/Shared/WebCoreArgumentCoders.h:
* Source/WebKit/Shared/libwpe/NativeWebTouchEventLibWPE.cpp:
* Source/WebKit/Shared/libwpe/WebEventFactory.cpp:
(WebKit::WebEventFactory::createWebTouchEvent):
* Source/WebKit/Shared/linux/WebMemorySamplerLinux.cpp:
(WebKit::nextToken):
* Source/WebKit/Shared/skia/CoreIPCSkColorSpace.h:
(WebKit::CoreIPCSkColorSpace::dataReference const):
* Source/WebKit/Shared/skia/CoreIPCSkData.h:
(WebKit::CoreIPCSkData::dataReference const):
* Source/WebKit/Shared/unix/AuxiliaryProcessMain.cpp:
* Source/WebKit/UIProcess/API/C/glib/WKTextCheckerGLib.cpp:
(WKTextCheckerSetSpellCheckingLanguages):
* Source/WebKit/UIProcess/API/glib/WebKitContextMenuClient.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitFaviconDatabase.cpp:
(webkitFaviconDatabaseGetFaviconInternal):
* Source/WebKit/UIProcess/API/glib/WebKitFileChooserRequest.cpp:
(webkit_file_chooser_request_select_files):
* Source/WebKit/UIProcess/API/glib/WebKitInitialize.cpp:
(WebKit::initializeRemoteInspectorServer):
* Source/WebKit/UIProcess/API/glib/WebKitProtocolHandler.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitURISchemeRequest.cpp:
(webkitURISchemeRequestReadCallback):
* Source/WebKit/UIProcess/API/glib/WebKitUserContent.cpp:
(toStringVector):
* Source/WebKit/UIProcess/API/glib/WebKitUserContentFilterStore.cpp:
(webkit_user_content_filter_store_fetch_identifiers):
* Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp:
(webkit_web_view_load_html):
(webkit_web_view_load_alternate_html):
(webkitWebViewEvaluateJavascriptInternal):
(webkitWebViewCallAsyncJavascriptFunctionInternal):
* Source/WebKit/UIProcess/API/glib/WebKitWebViewSessionState.cpp:
(decodeSessionState):
* Source/WebKit/UIProcess/API/gtk/DropTargetGtk4.cpp:
(WebKit::DropTarget::accept):
* Source/WebKit/UIProcess/API/gtk/WebKitInputMethodContextImplGtk.cpp:
(webkitInputMethodContextImplGtkNotifySurrounding):
* Source/WebKit/UIProcess/API/gtk/WebKitRemoteInspectorProtocolHandler.cpp:
(WebKit::RemoteInspectorProtocolHandler::updateTargetList):
* Source/WebKit/UIProcess/API/gtk/WebKitWebViewBase.cpp:
(shouldForwardWheelEvent):
* Source/WebKit/UIProcess/API/wpe/WPEWebViewPlatform.cpp:
* Source/WebKit/UIProcess/Automation/skia/WebAutomationSessionSkia.cpp:
(WebKit::base64EncodedPNGData):
* Source/WebKit/UIProcess/Inspector/glib/RemoteInspectorClient.cpp:
* Source/WebKit/UIProcess/Inspector/gtk/WebKitInspectorWindow.cpp:
* Source/WebKit/UIProcess/Launcher/glib/BubblewrapLauncher.cpp:
* Source/WebKit/UIProcess/Launcher/glib/FlatpakLauncher.cpp:
* Source/WebKit/UIProcess/Launcher/glib/ProcessLauncherGLib.cpp:
(WebKit::ProcessLauncher::launchProcess):
* Source/WebKit/UIProcess/Launcher/glib/XDGDBusProxy.cpp:
(WebKit::XDGDBusProxy::launch):
* Source/WebKit/UIProcess/Notifications/glib/NotificationService.cpp:
* Source/WebKit/UIProcess/glib/DisplayVBlankMonitorDRM.cpp:
* Source/WebKit/UIProcess/glib/WebProcessPoolGLib.cpp:
(WebKit::WebProcessPool::setSandboxEnabled):
* Source/WebKit/UIProcess/gtk/ClipboardGtk4.cpp:
(WebKit::Clipboard::formats):
* Source/WebKit/UIProcess/wpe/WebPageProxyWPE.cpp:
(WebKit::WebPageProxy::preferredBufferFormats const):
* Source/WebKit/WebProcess/GPU/webrtc/LibWebRTCCodecs.cpp:
* Source/WebKit/WebProcess/GPU/webrtc/SharedVideoFrame.cpp:
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitScriptWorld.cpp:
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebProcessExtensionManager.cpp:
(WebKit::parseUserData):
* Source/WebKit/WebProcess/Network/webrtc/LibWebRTCDnsResolverFactory.h:
* Source/WebKit/WebProcess/Network/webrtc/LibWebRTCProvider.cpp:
* Source/WebKit/WebProcess/Network/webrtc/LibWebRTCProvider.h:
* Source/WebKit/WebProcess/Network/webrtc/LibWebRTCSocket.h:
* Source/WebKit/WebProcess/Network/webrtc/LibWebRTCSocketFactory.h:
* Source/WebKit/WebProcess/WebCoreSupport/wpe/WebEditorClientWPE.cpp:
* Source/WebKit/WebProcess/WebPage/gtk/WebPrintOperationGtk.cpp:
(WebKit::WebPrintOperationGtk::PrintPagesData::PrintPagesData):
* Source/WebKit/WebProcess/WebPage/gtk/WebPrintOperationGtk.h:

Canonical link: <a href="https://commits.webkit.org/286279@main">https://commits.webkit.org/286279@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1c44fd7c555cb6e64e87e48fe319dc27fb39c602

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/75383 "7 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/45 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/28233 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/79858 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/26647 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/45 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/2614 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/59162 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/17381 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/78450 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/18 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/64765 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/39522 "Passed tests") | 
| | [⏳ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/iOS-17-Simulator-WPT-WK2-Tests-EWS "Waiting to run tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/22260 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/24974 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/68533 "Built successfully and passed tests") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/67755 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/22597 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/81339 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/74645 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/2722 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/1710 "Found 1 new test failure: fast/css/view-transitions-hide-under-page-background-color.html (failure)") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/67407 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/2873 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/64751 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/66687 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/10642 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/8802 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/96913 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/11648 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/2679 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/5500 "Built successfully") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/21170 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/2704 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/3639 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/2711 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

mcatanzaro commented on Source/WTF/wtf/Compiler.h:
```diff
@@ -641,3 +641,19 @@
+    WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN \
+    ALLOW_COMMA_BEGIN \
+    ALLOW_DEPRECATED_DECLARATIONS_BEGIN \
+    IGNORE_CLANG_WARNINGS_BEGIN("cast-align") \
```
```suggestion
    IGNORE_WARNINGS_BEGIN("cast-align") \
```

You surely want to suppress this warning when compiling with GCC as well.

mcatanzaro commented on Source/WTF/wtf/Compiler.h:
```diff
@@ -641,3 +641,19 @@
+    ALLOW_COMMA_END \
+    ALLOW_DEPRECATED_DECLARATIONS_END \
+    IGNORE_CLANG_WARNINGS_END \
+    IGNORE_CLANG_WARNINGS_END
```
If you take my suggestion and use `IGNORE_CLANG_WARNINGS_BEGIN` and `IGNORE_WARNINGS_BEGIN` together, then the order of those two will matter, so:

```suggestion
#define WTF_IGNORE_WARNINGS_IN_THIRD_PARTY_CODE_END \
    IGNORE_CLANG_WARNINGS_END \
    IGNORE_WARNINGS_END \
    ALLOW_DEPRECATED_DECLARATIONS_END \
    ALLOW_COMMA_END \
    WTF_ALLOW_UNSAFE_BUFFER_USAGE_END \
```

mcatanzaro commented on Source/WebKit/NetworkProcess/cache/NetworkCacheDataGLib.cpp:
```diff
@@ -104,7 +104,7 @@ Data concatenate(const Data& a, const Data& b)
 
     size_t size = a.size() + b.size();
-    WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN
+    WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN // GTK port
```
```suggestion
    WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN // WPE/GTK ports
```

mcatanzaro commented on Source/WebKit/NetworkProcess/soup/WebSocketTaskSoup.cpp:
```diff
@@ -66,7 +66,7 @@ WebSocketTask::WebSocketTask(NetworkSocketChannel& channel, const WebCore::Resou
     , m_delayFailTimer(RunLoop::main(), this, &WebSocketTask::delayFailTimerFired)
 {
-    WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN
+    WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN // GTK port
```
```suggestion
    WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN // WPE/GTK ports
```

mcatanzaro commented on Source/WebKit/Platform/IPC/glib/ArgumentCodersGlib.h:
```diff
@@ -26,7 +26,7 @@
 #include <wtf/text/CString.h>
 
-WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN
+WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN // GTK port
```
```suggestion
WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN // WPE/GTK ports
```

mcatanzaro commented on Source/WebKit/Shared/API/glib/WebKitURIRequest.cpp:
```diff
@@ -32,7 +32,7 @@ enum {
 };
 
-WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN
+WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN // GTK port
```
I'll stop mentioning these. Basically unless the file ends in "Gtk.cpp" or is in a gtk/ directory, it's probably shared by WPE as well.

mcatanzaro reviewed (commented):
Let's make sure to test with clang 19 before landing. I will add a to-do for myself to do a test build.

geoffreygaren committed:
Resolved the remaining uses of WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN in /WebKit/ in Apple ports
https://bugs.webkit.org/show_bug.cgi?id=282730
rdar://139395062

Reviewed by Chris Dumez.

You can now audit WebKit by searching for the regex

    WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN(?! //.*port)

and it should turn up no results.

Fixed one or two remaining unsafe buffer uses.

Added a WTF_IGNORE_WARNINGS_IN_THIRD_PARTY_CODE_BEGIN/_END to distinguish code
that we're skipping because it's not a part of WebKit. (Clang treats system
headers similarly.)

Added comments to annotate port-specific uses of
WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN to facilitate regex audit.

Removed some markers that skipped constructs that are now demonstrably safe.

* Source/WTF/wtf/Compiler.h:
* Source/WebKit/GPUProcess/EntryPoint/Cocoa/XPCService/GPUServiceEntryPoint.mm:
(GPU_SERVICE_INITIALIZER):
* Source/WebKit/GPUProcess/webrtc/LibWebRTCCodecsProxy.mm:
* Source/WebKit/NetworkProcess/cache/NetworkCacheDataGLib.cpp:
(WebKit::NetworkCache::concatenate):
* Source/WebKit/NetworkProcess/soup/WebSocketTaskSoup.cpp:
(WebKit::WebSocketTask::WebSocketTask):
* Source/WebKit/NetworkProcess/webrtc/LibWebRTCSocketClient.h:
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCProvider.cpp:
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCProvider.h:
* Source/WebKit/NetworkProcess/webrtc/NetworkRTCTCPSocketCocoa.mm:
* Source/WebKit/Platform/IPC/JSIPCBinding.h:
* Source/WebKit/Platform/IPC/StreamConnectionEncoder.h:
* Source/WebKit/Platform/IPC/glib/ArgumentCodersGlib.h:
* Source/WebKit/Platform/IPC/unix/ConnectionUnix.cpp:
* Source/WebKit/Platform/IPC/unix/UnixMessage.h:
(IPC::UnixMessage::UnixMessage):
* Source/WebKit/Shared/API/c/skia/WKImageSkia.h:
* Source/WebKit/Shared/API/glib/WebKitURIRequest.cpp:
(webkit_uri_request_set_uri):
* Source/WebKit/Shared/EntryPointUtilities/Cocoa/XPCService/XPCServiceEntryPoint.h:
* Source/WebKit/Shared/EntryPointUtilities/Cocoa/XPCService/XPCServiceEntryPoint.mm:
(WebKit::setJSCOptions):
* Source/WebKit/Shared/RTCNetwork.h:
* Source/WebKit/Shared/RTCPacketOptions.h:
* Source/WebKit/Shared/RTCWebKitEncodedFrameInfo.h:
* Source/WebKit/Shared/WebCoreArgumentCoders.h:
* Source/WebKit/Shared/libwpe/NativeWebTouchEventLibWPE.cpp:
* Source/WebKit/Shared/libwpe/WebEventFactory.cpp:
(WebKit::WebEventFactory::createWebTouchEvent):
* Source/WebKit/Shared/linux/WebMemorySamplerLinux.cpp:
(WebKit::nextToken):
* Source/WebKit/Shared/skia/CoreIPCSkColorSpace.h:
(WebKit::CoreIPCSkColorSpace::dataReference const):
* Source/WebKit/Shared/skia/CoreIPCSkData.h:
(WebKit::CoreIPCSkData::dataReference const):
* Source/WebKit/Shared/unix/AuxiliaryProcessMain.cpp:
* Source/WebKit/UIProcess/API/C/glib/WKTextCheckerGLib.cpp:
(WKTextCheckerSetSpellCheckingLanguages):
* Source/WebKit/UIProcess/API/glib/WebKitContextMenuClient.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitFaviconDatabase.cpp:
(webkitFaviconDatabaseGetFaviconInternal):
* Source/WebKit/UIProcess/API/glib/WebKitFileChooserRequest.cpp:
(webkit_file_chooser_request_select_files):
* Source/WebKit/UIProcess/API/glib/WebKitInitialize.cpp:
(WebKit::initializeRemoteInspectorServer):
* Source/WebKit/UIProcess/API/glib/WebKitProtocolHandler.cpp:
* Source/WebKit/UIProcess/API/glib/WebKitURISchemeRequest.cpp:
(webkitURISchemeRequestReadCallback):
* Source/WebKit/UIProcess/API/glib/WebKitUserContent.cpp:
(toStringVector):
* Source/WebKit/UIProcess/API/glib/WebKitUserContentFilterStore.cpp:
(webkit_user_content_filter_store_fetch_identifiers):
* Source/WebKit/UIProcess/API/glib/WebKitWebView.cpp:
(webkit_web_view_load_html):
(webkit_web_view_load_alternate_html):
(webkitWebViewEvaluateJavascriptInternal):
(webkitWebViewCallAsyncJavascriptFunctionInternal):
* Source/WebKit/UIProcess/API/glib/WebKitWebViewSessionState.cpp:
(decodeSessionState):
* Source/WebKit/UIProcess/API/gtk/DropTargetGtk4.cpp:
(WebKit::DropTarget::accept):
* Source/WebKit/UIProcess/API/gtk/WebKitInputMethodContextImplGtk.cpp:
(webkitInputMethodContextImplGtkNotifySurrounding):
* Source/WebKit/UIProcess/API/gtk/WebKitRemoteInspectorProtocolHandler.cpp:
(WebKit::RemoteInspectorProtocolHandler::updateTargetList):
* Source/WebKit/UIProcess/API/gtk/WebKitWebViewBase.cpp:
(shouldForwardWheelEvent):
* Source/WebKit/UIProcess/API/wpe/WPEWebViewPlatform.cpp:
* Source/WebKit/UIProcess/Automation/skia/WebAutomationSessionSkia.cpp:
(WebKit::base64EncodedPNGData):
* Source/WebKit/UIProcess/Inspector/glib/RemoteInspectorClient.cpp:
* Source/WebKit/UIProcess/Inspector/gtk/WebKitInspectorWindow.cpp:
* Source/WebKit/UIProcess/Launcher/glib/BubblewrapLauncher.cpp:
* Source/WebKit/UIProcess/Launcher/glib/FlatpakLauncher.cpp:
* Source/WebKit/UIProcess/Launcher/glib/ProcessLauncherGLib.cpp:
(WebKit::ProcessLauncher::launchProcess):
* Source/WebKit/UIProcess/Launcher/glib/XDGDBusProxy.cpp:
(WebKit::XDGDBusProxy::launch):
* Source/WebKit/UIProcess/Notifications/glib/NotificationService.cpp:
* Source/WebKit/UIProcess/glib/DisplayVBlankMonitorDRM.cpp:
* Source/WebKit/UIProcess/glib/WebProcessPoolGLib.cpp:
(WebKit::WebProcessPool::setSandboxEnabled):
* Source/WebKit/UIProcess/gtk/ClipboardGtk4.cpp:
(WebKit::Clipboard::formats):
* Source/WebKit/UIProcess/wpe/WebPageProxyWPE.cpp:
(WebKit::WebPageProxy::preferredBufferFormats const):
* Source/WebKit/WebProcess/GPU/webrtc/LibWebRTCCodecs.cpp:
* Source/WebKit/WebProcess/GPU/webrtc/SharedVideoFrame.cpp:
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitScriptWorld.cpp:
* Source/WebKit/WebProcess/InjectedBundle/API/glib/WebProcessExtensionManager.cpp:
(WebKit::parseUserData):
* Source/WebKit/WebProcess/Network/webrtc/LibWebRTCDnsResolverFactory.h:
* Source/WebKit/WebProcess/Network/webrtc/LibWebRTCProvider.cpp:
* Source/WebKit/WebProcess/Network/webrtc/LibWebRTCProvider.h:
* Source/WebKit/WebProcess/Network/webrtc/LibWebRTCSocket.h:
* Source/WebKit/WebProcess/Network/webrtc/LibWebRTCSocketFactory.h:
* Source/WebKit/WebProcess/WebCoreSupport/wpe/WebEditorClientWPE.cpp:
* Source/WebKit/WebProcess/WebPage/gtk/WebPrintOperationGtk.cpp:
(WebKit::WebPrintOperationGtk::PrintPagesData::PrintPagesData):
* Source/WebKit/WebProcess/WebPage/gtk/WebPrintOperationGtk.h:

Canonical link: https://commits.webkit.org/286279@main
